### DISCUSSION TITLE
"ip" query not showing the ip with user-agent plugin enabled

### DISCUSSION THREAD

**Discussion by KraiiFox** [searxng/searxng#2994] [Q&A] [answered]
Hello
I'm having trouble getting it to show my ip. With the "user-agent" plugin enabled, when I search for "ip" it doesn't show my ip address or anything. But when I search for "user agent" it shows my user agent just fine and it works on other public instances.

Other instances:
![Screenshot 2023-11-08 224752](https://github.com/searxng/searxng/assets/43300558/f317b259-f4d8-4062-b720-d14ff6fb2cea)

My instance:
![Screenshot 2023-11-08 at 22-47-24 f0x search](https://github.com/searxng/searxng/assets/43300558/037217c5-f3a9-4bd4-a935-9004b45f3267)

I'm not sure if I have misconfigured something in my settings or somewhere else.
Thanks!

**Comment by return42** [comment]
In your HTTP server, did you set ``X-Forwarded-For`` correctly .. FYI to determine the IP, the plugin user-agent uses the same function as the limiter --> https://docs.searxng.org/admin/searx.limiter.html

**Reply by KraiiFox** [reply]
Yeah I think it is configured correctly as my other docker containers are getting the real IP. 
Does it have something to do with the fact that the server is behind cloudflare?

I just checked the logs and it seems to be throwing an error for the plugin
```
Traceback (most recent call last):
  File "/usr/local/searxng/searx/plugins/__init__.py", line 205, in call
    ret = getattr(plugin, plugin_type)(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/searx/plugins/self_info.py", line 25, in post_search
    ip = get_real_ip(request)
         ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/searx/botdetection/_helpers.py", line 110, in get_real_ip
    x_for: int = cfg['real_ip.x_for']  # type: ignore
                 ~~~^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
```

**Reply by return42** [reply, marked as answer]
Thanks for the traceback .. its a bug in SearXNG .. I opened https://github.com/searxng/searxng/issues/2995 .. I think we can now close this discussion and go further with the issue #2995
